src/client/pfs_file.go

```diff
@@ -84,7 +78,7 @@ func (c APIClient) WithModifyFileClient(repo, commit string, cb func(*ModifyFile
 }
```

Comment:
We need to close regardless of the error right? Is it a leak otherwise?

Reply:
We don't want to close because that would indicate success. I'll add a cancel context so we clean up resources immediately.

Reply:
Actually, this unfortunately needs to be handled through the pach client context. So, to do this properly you would have to do `pachClient := pachClient.WithCtx(cancelCtx)` to get the appropriate cleanup upon failure. I think these semantics make sense as compared to having an abort function, and they line up with the semantics of the underlying gRPC. It does suck that you cannot just pass in a context for the call though. We have discussed taking the context out of the pach client, but that might be a large undertaking.

Reply:
We could use this pattern under the hood for the scoped calls, but the unscoped ones will need to be handled by the user.

Reply:
Most of the obj.Clients use the context for cancellation too.  Maybe we should take a context and mark it as TODO.  As long as it doesn't leak resources in practice, I think it's fine.